.github/workflows/precommit.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Precommit tests"
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/precommit.yaml'
+      - 'llvm/**'
+
+jobs:
+  build-llvm:
```

Comment:
The job name should also be something like `build-llvm-linux` if we are planning a separate job for Windows.  The job name is the key that's used when turning on required status checks, so it should be different from the job name used for Windows.